Frameworks/CoreGraphics/DWriteWrapper.mm

```diff
@@ -0,0 +1,572 @@
+            *pPropertyContext->stretch = DWRITE_FONT_STRETCH_EXTRA_CONDENSED;
+        } else if (CFEqual(propertyToken, CFSTR("CONDENSED")) || CFEqual(propertyToken, CFSTR("NARROW")) ||
+                   CFEqual(propertyToken, CFSTR("COND"))) {
+            *pPropertyContext->stretch = DWRITE_FONT_STRETCH_CONDENSED;
+        } else if (CFEqual(propertyToken, CFSTR("SEMICONDENSED"))) {
+            *pPropertyContext->stretch = DWRITE_FONT_STRETCH_SEMI_CONDENSED;
+            // skip since this is the default
+            // } else if (CFEqual(propertyToken, CFSTR("NORMAL"))) {
+            //     *pPropertyContext->stretch = DWRITE_FONT_STRETCH_NORMAL;
+
+        } else if (CFEqual(propertyToken, CFSTR("SEMIEXPANDED"))) {
+            *pPropertyContext->stretch = DWRITE_FONT_STRETCH_SEMI_EXPANDED;
+        } else if (CFEqual(propertyToken, CFSTR("EXPANDED"))) {
+            *pPropertyContext->stretch = DWRITE_FONT_STRETCH_EXPANDED;
+        } else if (CFEqual(propertyToken, CFSTR("EXTRAEXPANDED"))) {
+            *pPropertyContext->stretch = DWRITE_FONT_STRETCH_EXTRA_EXPANDED;
+        } else if (CFEqual(propertyToken, CFSTR("ULTRAEXPANDED"))) {
+            *pPropertyContext->stretch = DWRITE_FONT_STRETCH_ULTRA_EXPANDED;
+
+            // skip since this is the default
+            // } else if (CFEqual(propertyToken, CFSTR("NORMAL"))) {
+            //     *pPropertyContext->style = DWRITE_FONT_STYLE_NORMAL;
+        } else if (CFEqual(propertyToken, CFSTR("OBLIQUE"))) {
+            *pPropertyContext->style = DWRITE_FONT_STYLE_OBLIQUE;
+        } else if (CFEqual(propertyToken, CFSTR("ITALIC"))) {
+            *pPropertyContext->style = DWRITE_FONT_STYLE_ITALIC;
+        }
+    };
+
+    CFSetApplyFunction(propertyTokens, initPropertyFromToken, &propertyContext);
+}
+
+/**
+ * Creates a standard DWrite factory object
+ */
+HRESULT _DWriteCreateFactoryInstance(IDWriteFactory** outFactory) {
+    return DWriteCreateFactory(DWRITE_FACTORY_TYPE_SHARED, __uuidof(IDWriteFactory), reinterpret_cast<IUnknown**>(outFactory));
+}
+
+/**
+ * Creates a DWrite text format object using the system font collection
+ */
+HRESULT _DWriteCreateTextFormat(const wchar_t* fontFamilyName,
+                                DWRITE_FONT_WEIGHT weight,
+                                DWRITE_FONT_STYLE style,
+                                DWRITE_FONT_STRETCH stretch,
+                                float fontSize,
+                                IDWriteTextFormat** outTextFormat) {
+    // Get the direct write factory instance
+    ComPtr<IDWriteFactory> dwriteFactory;
+    HRESULT hr = _DWriteCreateFactoryInstance(&dwriteFactory);
+    if (FAILED(hr)) {
+        TraceError(TAG, L"Unable to create DWrite factory");
+        return hr;
+    }
+
+    return dwriteFactory
+        ->CreateTextFormat(fontFamilyName, NULL, weight, style, stretch, fontSize, __GetUserDefaultLocaleName().data(), outTextFormat);
+}
```

Comment:
?

---
In reply to: [83338764](https://github.com/Microsoft/WinObjC/pull/1145#discussion_r83338764) [](ancestors = 83338764)